### ISSUE TITLE
Proper ways for Multiline String and String Interpolation

### ISSUE THREAD

**Issue by swety2003** [b3b00/csly#481]
Is there a suitable way to implement multi-line string and string interpolation like C #?

**Comment by b3b00** [comment]
hello @swety2003 , using [GenericLexer](https://github.com/b3b00/csly/wiki/GenericLexer) , the [String](https://github.com/b3b00/csly/wiki/GenericLexer#strings) lexeme should natively handle multiline strings.

Regarding string interpolation, I think the best way is using [GenericLexer modes](https://github.com/b3b00/csly/wiki/GenericLexerModes)

hope this would help

Olivier

**Comment by b3b00** [comment]
@swety2003 are you ok with my answer ?
Could you close thé issue please ?

**Comment by swety2003** [comment]
Well, GenericLexer can indeed solve the problem of multi line strings, but the delimiter can only be one character. How can we parse strings with multiple characters as delimiters？
```C#
    [String(delimiter:"\"\"")]
    STRING,
```
LEXER_STRING_DELIMITER_MUST_BE_1_CHAR [FATAL] bad lexem "" :  StringToken lexeme delimiter char <STRING> must be 1 character length

**Comment by b3b00** [comment]
@swety2003 could you please explain precisely what kind of token you are trying to parse ?

**Comment by swety2003** [comment]
For example:
```
configureFlags = ""
  -system-zlib -system-libpng -system-libjpeg
  ${if openglSupport then "-dlopen-opengl
    -L${mesa}/lib -I${mesa}/include
    -L${libXmu}/lib -I${libXmu}/include" else ""}
  ${if threadSupport then "-thread" else "-no-thread"}
"";
```

**Comment by b3b00** [comment]
So, to sum up you need a string that
- starts and ends with double double quotes ""...""
- Can contain interpolated expressions surrounded by braces {}. Those expressions may contains strings, with one double quote as delimiter 

Then the [generic lexer modes](https://github.com/b3b00/csly/wiki/GenericLexerModes) is the way to go.

**Comment by swety2003** [comment]
OK, thank you for your help, I'll try to use it to solve the problem.

**Comment by b3b00** [comment]
You Can take a look at https://github.com/b3b00/csly/tree/dev/src/samples/SimpleTemplate for inspiration.

**Comment by b3b00** [comment]
@swety2003 , to fix issue #495 i've done what you want in the IndentedwhileParser. you can take a look it should answer most of your need
